### PR TITLE
Feature/mttl 1743

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-1743

## Describe this PR

* Implement processing for AccountCreated event
* Refactored ApiGateways and E2E testing of them.
* Used Hackney.Shared.Tenure package rather than local implementation of Tenure objects
* Used Hackney.Core.Sns package for message definition

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: AssetInformationListener.Tests/E2ETests/Steps/AccountCreatedUseCaseSteps.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+using AssetInformationListener.Infrastructure;
+using AssetInformationListener.Infrastructure.Exceptions;
+using FluentAssertions;
+using Hackney.Shared.Tenure.Boundary.Response;
+using System;
+using System.Threading.Tasks;
+
+namespace AssetInformationListener.Tests.E2ETests.Steps
+{
+    public class AccountCreatedUseCaseSteps : BaseSteps
```
Comment (humulla) [inline AssetInformationListener.Tests/E2ETests/Steps/AccountCreatedUseCaseSteps.cs:14]: I think the name should be `AccountCreatedStep`

File: terraform/production/main.tf [hunk 2]
```diff
@@ -44,6 +44,10 @@ data "aws_ssm_parameter" "tenure_sns_topic_arn" {
   name = "/sns-topic/production/tenure/arn"
 }
 
+data "aws_ssm_parameter" "accounts_sns_topic_arn" {
+  name = "/sns-topic/development/accounts/arn"
```
Comment (humulla) [inline terraform/production/main.tf:48]: Name should be `…/production/…`

File: terraform/staging/main.tf [hunk 3]
```diff
@@ -44,6 +44,10 @@ data "aws_ssm_parameter" "tenure_sns_topic_arn" {
   name = "/sns-topic/staging/tenure/arn"
 }
 
+data "aws_ssm_parameter" "accounts_sns_topic_arn" {
+  name = "/sns-topic/development/accounts/arn"
```
Comment (humulla) [inline terraform/staging/main.tf:48]: Name should be `../staging/…`

Comment (mikemoody-amido) [reply terraform/staging/main.tf:48]: Good call 👍

**Pull request merged**